mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestExplore.java

```diff
@@ -0,0 +1,192 @@
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.restapi.v2;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import javax.xml.bind.annotation.XmlElementWrapper;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.datamodel.common.MCRXMLMetadataManager;
+import org.mycore.frontend.jersey.MCRCacheControl;
+import org.mycore.restapi.v2.explore.model.MCRRestExploreResponse;
+import org.mycore.restapi.v2.explore.model.MCRRestExploreResponseObject;
+import org.mycore.solr.MCRSolrClientFactory;
+
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+
+/**
+ * This RestAPI endpoint provides simple access to the underlying SOLR server.
+ * It supports pagination, sorting and filtering.
+ * You can configure a global filter query and define solr fields, 
+ * which can be delivered as payload in the response.
+ * 
+ * @author Robert Stephan
+ *
+ */
+@Path("/explore")
+@OpenAPIDefinition(
+        tags = {})
+public class MCRRestExplore {
+    public static int MAX_ROWS = 1000;
+
+    @Context
+    Request request;
+
+    @Context
+    ServletContext context;
+
+    @GET
+    @Produces({ MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON + ";charset=UTF-8" })
+    @MCRCacheControl(maxAge = @MCRCacheControl.Age(time = 1, unit = TimeUnit.HOURS),
+                     sMaxAge = @MCRCacheControl.Age(time = 1, unit = TimeUnit.HOURS))
+    /*
+     * @Operation( summary = "Explore objects in this repository", responses
+     * = @ApiResponse( content = @Content(array = @ArraySchema(schema
+     * = @Schema(implementation = MCRObjectIDDate.class)))), tags =
+     * MCRRestUtils.TAG_MYCORE_OBJECT)
+     */
+    @XmlElementWrapper(
+            name = "mycoreobjects")
+    public Response exploreObjects(@QueryParam("start") String start, @QueryParam("rows") String rows,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'exploreObjects(String, String, String, List)' has an NPath complexity of 14400](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5542460)